src/utils.js

```diff
@@ -272,3 +272,72 @@ export function isEqualArray(arr1, arr2, cb) {
 
   return arr1.every(cb);
 }
+
+/**
+ * From Underscore.js:
+ * Returns a function, that, when invoked, will only be triggered at most once
+ * during a given window of time. Normally, the throttled function will run as
+ * much as it can, without ever going more than once per wait duration; but if
+ * you’d like to disable the execution on the leading edge, pass {leading: false}.
+ * To disable execution on the trailing edge, ditto.
+ * @param {Function} func
+ * @param {Number} wait     milliseconds
+ * @param {Object} options
+ *
+ * @return {Function}
+ */
+export function throttle(func, wait, options) {
+  let context;
+  let args;
+  let result;
+  let timeout = null;
+  let previous = 0;
+  if (!options) {
+    options = {};
+  }
+  let later = () => {
+    previous = options.leading === false ? 0 : Date.now();
+    timeout = null;
+    result = func.apply(context, args);
+    if (!timeout) {
+      context = args = null;
+    }
+  };
+  return () => {
+    let now = Date.now();
+    if (!previous && options.leading === false) {
+      previous = now;
+    }
+    let remaining = wait - (now - previous);
+    context = this;
+    args = arguments;
+    if (remaining <= 0 || remaining > wait) {
+      if (timeout) {
+        clearTimeout(timeout);
+        timeout = null;
+      }
+      previous = now;
+      result = func.apply(context, args);
+      if (!timeout) {
+        context = args = null;
+      }
+    } else if (!timeout && options.trailing !== false) {
+      timeout = setTimeout(later, remaining);
+    }
+    return result;
+  };
+}
+
+/**
+ * Polyfill for finding document scrollTop.
+ */
+export function getScrollTop() {
```

Comment:
Could you please add some unit tests to this>

Reply:
test added